apps/theming-designer/src/components/AccessibilityDetailsList.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import {
+  DetailsList,
+  DetailsRow,
+  IDetailsRowStyles,
+  IDetailsRowProps,
+  IColumn,
+  IGroup,
+  MessageBar,
+  MessageBarType,
+  SelectionMode,
+  ITheme
+} from '../../../../packages/office-ui-fabric-react/lib/index';
+import { IContrastRatioPair } from './AccessibilityChecker';
+
+export interface IAccessibilityDetailsListProps {
+  allContrastRatioPairs: IContrastRatioPair[];
+  nonAccessibleStartIndex: number;
+  theme: ITheme | undefined;
+}
+
+interface IAccessibilityDetailsList {
+  key: string;
+  contrastRatio: String;
+  slotPair: String;
+}
+
+export const AccessibilityDetailsList: React.StatelessComponent<IAccessibilityDetailsListProps> = (
+  props: IAccessibilityDetailsListProps
+) => {
+  let items: IAccessibilityDetailsList[] = [];
+  let groups: IGroup[] = [];
+  let columns: IColumn[] = [];
+  const newTheme = props.theme;
+
+  const onRenderRow = (detailsRowProps: IDetailsRowProps | undefined): JSX.Element => {
+    // Set each row's background and text color to what's specified by its respective slot rule
+    const currentSlotPair = detailsRowProps!.item.slotPair;
+    const pairSplt = currentSlotPair.split(' on ');
```

Comment:
Can we pass in the colors as objects instead of splitting the string?

Reply:
I tried to rearchitect a bunch of the data structures in AccessibilityChecker and AccessibilityDetailsList to make this happen but every path I took resulted in a dead end - for example, when I tried to make IContrastRatioPair taken in just IColors for background color & foreground color, it was hard to make the items in AccessibilityDetailsList get pushed the right slot pair string (conversion to string just resulted in the hex)...